src/libexpr/eval-cache.cc

```diff
@@ -599,7 +599,10 @@ string_t AttrCursor::getStringWithContext()
                             return d.drvPath;
                         },
                         [&](const NixStringContextElem::Built & b) -> const StorePath & {
-                            return b.drvPath;
+                            if (auto * drvPath = std::get_if<SingleDerivedPath::Opaque>(&*b.drvPath))
+                                return drvPath->path;
+                            else
+                                root->state.error("Dynamic derivations in string contexts are not yet supported by the evaluation cache").debugThrow<UnimplementedError>();
```

Comment:
```suggestion
                                root->state.error("Dynamic derivations in string contexts are not yet supported by the evaluation cache. Try with `--option eval-cache false` for now.").debugThrow<UnimplementedError>();
```

Reply:
Please add this to the tracking issue.

Reply:
Oh actually I think that is not true anymore :), since this PR updates `NixStringContextElem` (which is now structured, unlike when the PR was written) at the same time.